spec/faker/bot/reflectors/search_spec.rb

```diff
@@ -4,25 +4,48 @@
 
 RSpec.describe Faker::Bot::Reflectors::Search do
   describe '#call' do
-    context 'when a match is found' do
+    let(:result) { described_class.new(query).call }
```

Comment:
Interesting change here, it's definitely more reusable across the test examples. I've been trying out a different approach by defining verification logic in the test examples to mitigate the [Mystery Guest](https://thoughtbot.com/blog/mystery-guest) test smell.

To be fair, this is a very common testing pattern in the RSpec world, so I doubt many people will have trouble making sense of it. Happy to hear your thoughts on the same. 🙂

Reply:
That's a great article! I was trying to decide if I should follow the existing pattern or avoid duplicate code, since so much of it could be shared. I'm happy to adopt the testing style you've already implemented in this app moving forward, just let me know!